web/src/compose.js

```diff
@@ -215,6 +216,11 @@ export function send_message(request = create_message_object()) {
 
     const message = echo.try_deliver_locally(request);
     if (message) {
+        // It is a little bit funny to go through the message_events
+        // codepath, but it's sort of the idea behind local echo that
+        // we are simulating server events before they actually arrive.
+        message_events.insert_new_messages([message], true);
```

Comment:
So this move in where the code lives is potentially a fine choice, but you need to rename at least `insert_local_message`, since that function no longer attempts to insert the message! And possibly also `try_delivery_locally` too. 

That said, would it be better to just pass the insertion function to the `echo` library as a parameter, such that the API would be `echo.,try_deliver_locally(request, insert_new_message)`.

Reply:
Oh right I like the latter solution better we can just pass that function as a callback.